src/game/RandomizationOfShipsPositions.js

```diff
@@ -0,0 +1,149 @@
+    }
+  }
+};
+createFieldsName();
+
+const createGameBoardArray = () => {
+  gameBoardArray = [];
+  for (let i = 0; i < GAME_BOARD_WIDTH; i++) {
+    gameBoardArray.push([]);
+    for (let j = 0; j < GAME_BOARD_WIDTH; j++) {
+      gameBoardArray[i].push({
+        isEmpty: true,
+        shipName: "",
+      });
+    }
+  }
+};
+
+const rollDirection = () => {
+  return Math.floor(Math.random() * 2) === 0 ? "vertical" : "horizontal";
+};
+
+const createPosition = (shipIndex, direction) => {
+  position = [];
+  for (let i = 0; i < 2; i++) {
+    position.push(Math.floor(Math.random() * GAME_BOARD_WIDTH));
+  }
+  if (
+    direction === "horizontal" &&
+    position[1] + SHIPS_LIST[shipIndex].size > GAME_BOARD_WIDTH
+  ) {
+    position[1] = Math.abs(position[1] - SHIPS_LIST[shipIndex].size);
+  } else if (position[0] + SHIPS_LIST[shipIndex].size > GAME_BOARD_WIDTH) {
+    position[0] = Math.abs(position[0] - SHIPS_LIST[shipIndex].size);
+  }
+};
+
+const addShip = () => {
+  createGameBoardArray();
+  for (let shipIndex = 0; shipIndex < SHIPS_LIST.length; shipIndex++) {
+    const direction = rollDirection();
+    if (direction === "horizontal") {
+      positionShipHorizontally(shipIndex, direction);
+    } else {
+      positionShipVertically(shipIndex, direction);
+    }
+  }
+};
+
+const positionShipVertically = (shipIndex, direction) => {
+  createPosition(shipIndex, direction);
+  let positionsForCheck = [];
+  for (let i = 0; i < SHIPS_LIST[shipIndex].size; i++) {
+    positionsForCheck.push(gameBoardArray[position[0] + i][position[1]]);
+  }
+  if (positionsForCheck.some((el) => el.isEmpty === false)) {
+    positionShipVertically(shipIndex);
+  } else {
+    for (let i = 0; i < SHIPS_LIST[shipIndex].size; i++) {
+      gameBoardArray[position[0] + i].fill(
+        {
+          isEmpty: false,
+          shipName: SHIPS_LIST[shipIndex].shipName,
+        },
+        position[1],
+        position[1] + 1
+      );
+    }
+  }
+};
+
+const positionShipHorizontally = (shipIndex, direction) => {
+  createPosition(shipIndex, direction);
+  if (
+    gameBoardArray[position[0]]
+      .slice(position[1], position[1] + SHIPS_LIST[shipIndex].size)
+      .some((el) => el.isEmpty === false)
+  ) {
+    positionShipHorizontally(shipIndex);
+  } else {
+    gameBoardArray[position[0]].fill(
+      {
+        isEmpty: false,
+        shipName: SHIPS_LIST[shipIndex].shipName,
+      },
+      position[1],
+      position[1] + SHIPS_LIST[shipIndex].size
+    );
+  }
+};
+
+export const addShipsToBoard = () => {
+  addShip();
+  GAME_BOARD[0].innerHTML = "";
+  let gameBoardForMap = gameBoardArray.flat(Infinity);
+  gameBoardForMap.map((_, i) => {
+    let newField = document.createElement("div");
+    newField.classList.add(`board__field`);
+    if (gameBoardForMap[i].isEmpty === false) {
+      newField.classList.add(`ship`);
+    }
+    if (gameBoardForMap[i].shipName != "") {
```

Comment:
Just in case if you didn't know - strings have lengths so in an alternative you could use `gameBoardForMap[i].shipName.length > 0` - it is just a piece of information you can do whatever you want with it 😄